lib/px/service/kinesis/base_request.rb

```diff
@@ -43,32 +44,41 @@ def flush_records
       #
       @buffer = @buffer.compact
       if @buffer.present? && can_flush?
-        begin
-          response = @kinesis.put_records(stream_name: @stream, records: @buffer)
-        rescue Exception => e
-          puts "[#{DateTime.now}] Raised error with input: #{@buffer} with error #{e}"
-          raise
-        end
 
-        # iterate over response and
-        # back append everything that didn't send
-        #
-        # TODO: detect which shard is being limited
-        # - decay that shard's partition_key
-        # - split shards when we really need to
-
-        tmp_buffer = []
-        if response[:failed_record_count] > 0
-          response[:records].each_with_index do |r, index|
-            if r.error_code == Aws::Kinesis::Errors::ProvisionedThroughputExceededException.code
-              # set last throughput limited value
-              @last_throughput_exceeded = Time.now
+        if Rails.env.development?
```

Comment:
`Rails` won't always be defined (e.g. the BFFs aren't Rails apps).  Use a config setting instead.